src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -1892,6 +1893,48 @@ export function registerTerminalActions() {
 			accessor.get(ITerminalService).doWithActiveInstance(t => t.clearBuffer());
 		}
 	});
+	registerAction2(class extends Action2 {
+		constructor() {
+			super({
+				id: TerminalCommandId.ShowWordLinkQuickpick,
+				title: { value: localize('workbench.action.terminal.showWordLinkQuickpick', "Show Word Link Quick Pick"), original: 'Show Word Link Quickpick' },
+				f1: true,
+				category,
+				precondition: ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated),
+			});
+		}
+		run(accessor: ServicesAccessor) {
+			accessor.get(ITerminalService).doWithActiveInstance(t => t.showLinkQuickpick(TerminalLinkProviderType.Word));
+		}
+	});
+	registerAction2(class extends Action2 {
+		constructor() {
+			super({
+				id: TerminalCommandId.ShowProtocolLinkQuickpick,
+				title: { value: localize('workbench.action.terminal.showProtocolLinkQuickpick', "Show Protocol Link Quick Pick"), original: 'Show Protocol Link Quickpick' },
+				f1: true,
+				category,
+				precondition: ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated),
+			});
+		}
+		run(accessor: ServicesAccessor) {
+			accessor.get(ITerminalService).doWithActiveInstance(t => t.showLinkQuickpick(TerminalLinkProviderType.Protocol));
+		}
+	});
+	registerAction2(class extends Action2 {
+		constructor() {
+			super({
+				id: TerminalCommandId.ShowValidatedLinkQuickpick,
```

Comment:
How about a general `Terminal: Open Detected Link` which shows all link providers, and that allows filtering by type? Not sure the format but with this we should be able to trigger a keybinding that opens only links from gitlens for example.

Reply:
How about `Terminal: Show Detected Links` because selecting a word will copy, not open, it

Reply:
Oh I missed that, I think it should still open it just as it would normally be via ctrl+click. So it will open if there is exactly one match, otherwise act as a shortcut to ctrl+p, <word>